test/e2e/antreapolicy_test.go

```diff
@@ -108,19 +90,82 @@ type podToAddrTestStep struct {
 	expectedConnectivity PodConnectivityMark
 }
 
-func initialize(t *testing.T, data *TestData) {
+// Util function to get the runtime name of a test Namespace.
+func getNS(ns string) string {
+	return namespaces[ns].Name
+}
+
+// Util function to get the runtime Pod struct of a test Pod.
+func getPod(ns, po string) Pod {
+	return Pod(namespaces[ns].Name + "/" + po)
+}
+
+// Util function to get the runtime Pod name of a test Pod.
+func getPodName(ns, po string) string {
+	return namespaces[ns].Name + "/" + po
+}
+
+// initNamespaceMeta populates the test Namespaces metadata.
+// There are two form factors for test workload Namespaces:
+//
+//	Normal: three Namespaces x, y, z.
+//	Large: two "prod" Namespaces labeled purpose=test and tier=prod.
+//	       two "dev" Namespaces labeled purpose=test and tier=dev.
+//	       one "no-tier-label" Namespace labeled purpose=test.
+//
+// The large form factor workloads are used for testcases where advanced
+// Namespace matching in policies are required.
+func initNamespaceMeta(formFactor string) map[string]TestNamespaceMeta {
```

Comment:
I'm wondering whether it will be easier if we just add labels needed by this feature to exist x y z NS and revert them after the test case.

Reply:
Not necessarily in my opinion, as we need to test all different groups and namespace labels for the `sameLabels` feature. 3 is not enough.

Reply:
I'm also hoping with the new TestNamespaceMeta, the overall antreapolicy_test code can become cleaner. A lot of the original code was adopted from a reachability poc which was kind of hacky and not easier to maintain